manifests/profile/rabbitmq.pp

```diff
@@ -10,10 +10,35 @@
     }
   }
 
-  class { '::nova::rabbitmq':
-    userid             => $::openstack::config::rabbitmq_user,
-    password           => $::openstack::config::rabbitmq_password,
-    cluster_disk_nodes => [$management_address],
-    rabbitmq_class     => '::rabbitmq',
+  # Ideally we would configure the firewall to open the rabbitmq ports
+  # and allow access to the rabbitmq nodes, but we can't restrict the
+  # clustered node ports with early version of the rabbitmq module and
+  # therefore we can't open those ports. Instead we just ensure that
+  # requests on localhost use the localhost network. We will replace
+  # this with firewall rules when all the stackforge modules have
+  # upgraded to puppetlabs-rabbitmq >=5.x
+  host { 'localhost':
+    ensure => present,
+    name   => $::hostname,
+    ip     => '127.0.0.1',
+  }
+
+  rabbitmq_user { $::openstack::config::rabbitmq_user:
+    admin     => true,
+    password  => $::openstack::config::rabbitmq_password,
+    provider  => 'rabbitmqctl',
+    require   => Class['::rabbitmq'],
+  }
+  rabbitmq_user_permissions { "${openstack::config::rabbitmq_user}@/":
+    configure_permission => '.*',
+    write_permission     => '.*',
+    read_permission      => '.*',
+    provider             => 'rabbitmqctl',
+  }->Anchor<| title == 'nova-start' |>
```

Comment:
Actually this probably shouldn't be necessary at all with all the relationships defined in the role, and doesn't really belong here since the profile is for rabbitmq and not nova. Will test to verify.